### ISSUE TITLE
Audio skills problems

### ISSUE THREAD

**Issue by mizansyed** [patst/alexa-skills-kit-for-go#20]
Firstly thanks for providing this package.
I am trying to use your package to build an Audio Skill but getting 

> There was a problem with the requested skill's response

I am trying to figure out what is causing this issue, but unsure where the problem lies. Am I missing something? Thanks for your help.

Here are snippets (removed original mp3 url)

```
responseEnvelope.Response.AddAudioPlayerClearQueueDirective("CLEAR_ALL")
play := responseEnvelope.Response.AddAudioPlayerPlayDirective("REPLACE_ALL")
play.SetAudioItemStream("https://someurl/file.mp3", "1234AAA", "", 0)
play.SetAudioItemMetadata("ABC", "Not available")`
```
The json output from the Alexa Simulator

```
{
	"body": {
		"version": "1.0",
		"response": {
			"directives": [
				{
					"type": "AudioPlayer.Play",
					"playBehavior": "ENQUEUE",
					"audioItem": {
						"stream": {
							"expectedPreviousToken": "1234AAAABBBBCCCCCDDDDEEEEEFFFFA",
							"token": "1234AAAABBBBCCCCCDDDDEEEEEFFFF",
							"url": "https://someurl/file.mp3",
							"offsetInMilliseconds": 0
						},
						"metadata": {
							"title": "ABC",
							"subtitle": "Not available",
							"art": {
								"contentDescription": ""
							},
							"backgroundImage": {
								"contentDescription": ""
							}
						}
					}
				}
			]
		}
	}
}

```

**Comment by patst** [comment]
Hey, 
thanks for the feedback and sorry for the inconvenience. I never implemented a audio skill.

I think the problem are the image objects in the metadata. (https://developer.amazon.com/docs/custom-skills/audioplayer-interface-reference.html#play ) 
Expected is a metadata object which should be looking this way:

```
"metadata": {
      "title": "title of the track to display",
      "subtitle": "subtitle of the track to display",
      "art": {
        "sources": [
          {
            "url": "https://url-of-the-album-art-image.png"
          }
        ]
      },
      "backgroundImage": {
        "sources": [
          {
            "url": "https://url-of-the-background-image.png"
          }
        ]
      }
```

The `sources` wrapper seems to be missing for the images. Apart from that I think the tags should not be there in the first place.

I'm trying to create a test case to reproduce the problem and create a fix for that.

**Comment by patst** [comment]
I tried to fix the problems in #22 .
The art and background image tags should no longer appear in the alexa simulator.


If you want to add image metadata for the echo show at a later point:

If you call SetAudioItemMetadata you now get a object back and are able to set the image sources information if you like. If you don't set it the response json will no longer contain the elements.

Create image:
```
play := response.Response.AddAudioPlayerPlayDirective("play")
metadata := play.SetAudioItemMetadata("title", "subtitle")
artImage := metadata.SetArtImage("artImage")
artImage.AddImageSource("MEDIUM", "http://....", 200, 300)
```

It would be great if you can test if that fixes your problem! It is merged in the upstream version

**Comment by mizansyed** [comment]
Thanks @patst. I had an opportunity to test today and it worked.
